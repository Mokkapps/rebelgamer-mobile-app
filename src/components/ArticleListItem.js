// @flow
import { Dimensions, Image, StyleSheet, Text, View } from 'react-native';
import React from 'react';

import Constants from './../constants';
import DateUtils from './../utils/DateUtils';
import HtmlDecoder from './../utils/HtmlDecoder';
import Post from './../types';

type Props = {
  article: Post
};

class ArticleListItem extends React.PureComponent<Props> {
  setNativeProps = (nativeProps: Props) => {
    if (this.root) {
      this.root.setNativeProps(nativeProps);
    }
  };

  root: View | null;

  render() {
    const { article } = this.props;
    return (
      // eslint-disable-next-line no-return-assign
      <View ref={component => (this.root = component)} {...this.props}>
        <Image
          style={styles.image}
          source={{ uri: article._embedded['wp:featuredmedia'][0].source_url }}
        />
        <Text style={styles.headline}>
          {HtmlDecoder.decodeHtml(article.title.rendered)}
        </Text>
        <Text style={styles.date}>
          {DateUtils.getPostedAtDateString(article.date)}
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center'
  },
  image: {
    width: Dimensions.get('window').width,
    height: Constants.HeadlineImageHeigth,
    marginBottom: 5
  },
  headline: {
    fontWeight: 'bold',
    color: 'black',
    fontSize: Constants.FontSizeHeadline,
    marginHorizontal: 10
  },
  date: {
    margin: 10,
    fontSize: Constants.FontSizeListDate
  }
});

export default ArticleListItem;
