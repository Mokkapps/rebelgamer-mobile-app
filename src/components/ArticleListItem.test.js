import React from 'react';
import renderer from 'react-test-renderer';

import ArticleListItem from './ArticleListItem';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <ArticleListItem
        article={{
          id: 124,
          title: { rendered: 'title' },
          _embedded: { 'wp:featuredmedia': [{ source_url: 'URL' }] }
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
