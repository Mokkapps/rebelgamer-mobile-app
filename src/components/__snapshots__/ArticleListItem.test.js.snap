// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  article={
    Object {
      "_embedded": Object {
        "wp:featuredmedia": Array [
          Object {
            "source_url": "URL",
          },
        ],
      },
      "id": 124,
      "title": Object {
        "rendered": "title",
      },
    }
  }
>
  <Image
    source={
      Object {
        "uri": "URL",
      }
    }
    style={
      Object {
        "height": 400,
        "marginBottom": 5,
        "width": 750,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
        "fontSize": 30,
        "fontWeight": "bold",
        "marginHorizontal": 10,
      }
    }
  >
    title
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 15,
        "margin": 10,
      }
    }
  >
    Vor kurzem
  </Text>
</View>
`;
